src/javascript/tests/lists.rs

```diff
@@ -82,3 +82,49 @@ const b = [1, [2, [3, []]]];
 "#
     );
 }
+
+#[test]
+fn list_destructuring() {
+    assert_js!(
+        r#"
+fn go(x, y) {
+    let [] = x
+    let [a] = x
+    let [1, 2] = x
+    let [_, #(3, b)] = y
+    let [head, ..tail] = y
+}
+"#,
+        r#""use strict";
+
+function go(x, y) {
+  let gleam$tmp = x;
+  if (!(gleam$tmp?.length === 0)) throw new Error("Bad match");
```

Comment:
Can shorten this using arrays truthy falseyness

```suggestion
  if (!(!gleam$tmp)) throw new Error("Bad match");
```

Reply:
Ah yes! That makes perfect sense. Thank you.